src/apps/frontend/components/error/error-boundary.tsx

```diff
@@ -0,0 +1,51 @@
+import axios from 'axios';
+import React, { Component, ErrorInfo, ReactNode } from 'react';
+
+import { JsonObject } from '../../types/common-types';
+
+interface Props {
+  children?: ReactNode;
+}
+
+interface State {
+  hasError: boolean;
+}
+
+class ErrorBoundary extends Component<Props, State> {
+  public state: State = {
+    hasError: false,
+  };
+
+  errorData: JsonObject & {
+    'error-info': string;
+    'error-message': string;
+    'error-name': string;
+  };
+
+  public static getDerivedStateFromError(): State {
+    return { hasError: true };
+  }
+
+  public async componentDidCatch(error: Error, errorInfo: ErrorInfo) {
+    this.errorData = {
+      'error-name': error.name,
+      'error-message': error.message,
+      'error-info': errorInfo.componentStack,
+    };
+    try {
+      await axios.post('http://127.0.0.1:8080/client_logs', this.errorData);
```

Comment:
Server would not run at this endpoint always, right? For example, in production, this will not work.